File: java/src/test/java/com/aliyun/container/service/cost/sdk/ClientTest.java
package com.aliyun.container.service.cost.sdk;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.Map;
import java.util.HashMap;

public class ClientTest {
    private Client client;
    private Config config;

    @Before
    public void setUp() {
        config = new Config();
        config.setEndpoint("http://127.0.0.1:8080");
        client = new Client(config);
    }

    @Test
    public void testClientCreation() {
        assertNotNull("Client should be created", client);
        assertNotNull("Config should be set", client.getConfig());
        assertEquals("Endpoint should match", "http://127.0.0.1:8080", client.getConfig().getEndpoint());
    }

    @Test
    public void testGetCostV2WithMockData() {
        Map<String, Object> request = new HashMap<>();
        request.put("window", "today");
        
        Map<String, Object> response = client.getCostV2(request);
        
        assertNotNull("Response should not be null", response);
        assertTrue("Response should contain code", response.containsKey("code"));
        // 由于使用本地地址，应该返回模拟数据
        assertEquals("Should return mock data with code 200", 200, response.get("code"));
        assertTrue("Should contain data", response.containsKey("data"));
    }

    @Test
    public void testGetAllocationWithMockData() {
        Map<String, Object> request = new HashMap<>();
        request.put("window", "today");
        
        Map<String, Object> response = client.getAllocation(request);
        
        assertNotNull("Response should not be null", response);
        assertTrue("Response should contain code", response.containsKey("code"));
        // 由于使用本地地址，应该返回模拟数据
        assertEquals("Should return mock data with code 200", 200, response.get("code"));
        assertTrue("Should contain data", response.containsKey("data"));
    }

    @Test
    public void testGetCostV2WithNamespaceAggregation() {
        Map<String, Object> request = new HashMap<>();
        request.put("window", "today");
        request.put("aggregate", "namespace");
        
        Map<String, Object> response = client.getCostV2(request);
        
        assertNotNull("Response should not be null", response);
        assertEquals("Should return mock data with code 200", 200, response.get("code"));
        assertTrue("Should contain data", response.containsKey("data"));
    }

    @Test
    public void testGetAllocationWithNamespaceAggregation() {
        Map<String, Object> request = new HashMap<>();
        request.put("window", "today");
        request.put("aggregate", "namespace");
        
        Map<String, Object> response = client.getAllocation(request);
        
        assertNotNull("Response should not be null", response);
        assertEquals("Should return mock data with code 200", 200, response.get("code"));
        assertTrue("Should contain data", response.containsKey("data"));
    }
}

File: javascript/test/client.test.js
const { Client, Config } = require('../src/client');

describe('Client', () => {
    let client;
    let config;

    beforeEach(() => {
        config = new Config();
        config.endpoint = 'http://127.0.0.1:8080';
        client = new Client(config);
    });

    test('should create client with config', () => {
        expect(client).toBeDefined();
        expect(client.config).toBeDefined();
        expect(client.config.endpoint).toBe('http://127.0.0.1:8080');
    });

    test('should create client without config', () => {
        const newClient = new Client();
        expect(newClient).toBeDefined();
        expect(newClient.config).toBeDefined();
        expect(newClient.config.endpoint).toBe('http://127.0.0.1:8080');
    });

    test('should return mock cost v2 data', async () => {
        const request = { window: 'today' };
        const response = await client.getCostV2(request);
        
        expect(response).toBeDefined();
        expect(response.code).toBe(200);
        expect(response.data).toBeDefined();
    });

    test('should return mock allocation data', async () => {
        const request = { window: 'today' };
        const response = await client.getAllocation(request);
        
        expect(response).toBeDefined();
        expect(response.code).toBe(200);
        expect(response.data).toBeDefined();
    });

    test('should return mock cost v2 data with namespace aggregation', async () => {
        const request = { 
            window: 'today',
            aggregate: 'namespace'
        };
        const response = await client.getCostV2(request);
        
        expect(response).toBeDefined();
        expect(response.code).toBe(200);
        expect(response.data).toBeDefined();
        expect(response.data.items).toBeDefined();
    });

    test('should return mock allocation data with namespace aggregation', async () => {
        const request = { 
            window: 'today',
            aggregate: 'namespace'
        };
        const response = await client.getAllocation(request);
        
        expect(response).toBeDefined();
        expect(response.code).toBe(200);
        expect(response.data).toBeDefined();
        expect(response.data.items).toBeDefined();
    });
});

describe('Config', () => {
    test('should create config with default values', () => {
        const config = new Config();
        expect(config.endpoint).toBe('http://127.0.0.1:8080');
        expect(config.accessKeyId).toBe('');
        expect(config.accessKeySecret).toBe('');
    });

    test('should allow setting endpoint', () => {
        const config = new Config();
        config.endpoint = 'http://test.endpoint.com';
        expect(config.endpoint).toBe('http://test.endpoint.com');
    });
});